### DISCUSSION TITLE
Azure Functions vs Azure VM - security

### DISCUSSION THREAD

**Discussion by marrybus** [AzureCommunityPL/group#272] [Pomoc]
hej,
potrzebuje w maksymalnie bezpieczny sposób przetworzyć dane z jednego bloba do innego - po drodze deszyfrując dane z pierwszego i po odpowiednich przekształceniach ponowne zaszyfrowanie w drugim.

Przyjmijmy że klucz jest przechowywany poza Azure w jakimś KMSie. W przypadku robienia tego za pomocą Azure VM przez czas całej operacji klucz deszyfrujący będzie w RAMie VMki. Jeżeli ktoś - np NSA złapie dumpa takiej VMki to może dobrać się do klucza. Jak sytuacja wygląda przy Azure Functions? czy będę w stanie uniknąć tego problemu? kto - jeżeli w ogóle -może złapać dumpa pamięci wykonywanych zadań? 

Pozdrawiam
Mariusz

**Comment by kamil-mrzyglod** [comment]
Tak szczerze to jeśli jest to dla Ciebie jakiś wektor ataku(dump pamięci i wyciągnięcie klucza) to chyba nie do końca problemem jest to, czego używasz(VM vs AF) tylko jak implementacyjnie to robisz. AF też ma gdzieś pod spodem VMkę tylko Ty jej już nie widzisz.

Pamiętam, że dla niektórych klientów rozwiązaniem było dokonanie deszyfracji danych "offline" - czyli download zaszyfrowanych danych, decrypt lokalnie, processing, encrypt, upload. W innych projektach uwaga była skupiona bardziej na użyciu bezpiecznych struktur w celu zminimalizowania ryzyka(typu SecureString w .NET, który trochę udawał realne zabezpieczenie albo przechowywanie sekretów w pamięci używających czegoś innego niż prosty łańcuch znaków). Generalnie problem jest bardziej związany z walającym się po RAMie stringiem przez cały okres działania programu niż wyborem konkretnej usługi ;)

**Comment by piotrgwiazda** [comment]
Taki model ochrony level "insane" jest wspierany tylko przez Azure
Confidential Compute
https://azure.microsoft.com/en-us/solutions/confidential-compute/

Generalnie jednak zgadzam się z przedmówcą.

wt., 27 kwi 2021, 16:54 użytkownik Kamil Mrzygłód ***@***.***>
napisał:

**Reply by kamil-mrzyglod** [reply]
O, coś mi po głowie chodziło, ale nie mogłem sobie przypomnieć jak to się w Azure nazywa. Tylko to chyba obecnie Private Preview?

**Comment by marrybus** [comment]
dzieki za hinty - zobaczymy co klient na to
